Daily tests

yarondbb opened the pull request.

yarondbb committed:
Added application file

yarondbb committed:
Added application file

yarondbb committed:
Added unit tests methods

yarondbb committed:
.

yarondbb committed:
added support for path for unix (under /tmp) and windows (under user dir)

yarondbb committed:
.

yarondbb committed:
for rake, do not raise error when 'error' string is returned from commnad since it does not necessarity indicated a real error but rather an indiation that no error had occurred

yarondbb committed:
for rake, do not raise error when 'error' string is returned from commnad since it does not necessarity indicated a real error but rather an indiation that no error had occurred

yarondbb committed:
Added email suppport

yarondbb committed:
mail support

yarondbb committed:
Added main log support

yarondbb committed:
.

yarondbb committed:
.

yarondbb committed:
.

yarondbb committed:
Added scheduling code (start at midnight)

yarondbb committed:
fixed time format for file name

yarondbb committed:
support many executions in 1 day. This will make different dirs

yarondbb committed:
fix dirs path issue

yarondbb committed:
.

bbfsdev commented on bin/daily_tests.rb:
```diff
@@ -0,0 +1,221 @@
+# ------------------------------------------------- Definitions ---------------------------
+BBFS_GIT_REPO = 'https://github.com/bbfsdev/bbfs'
+
+FROM_EMAIL = ARGV[0]
```
You can use the params module if needed.

bbfsdev commented on bin/daily_tests.rb:
```diff
@@ -0,0 +1,221 @@
+    chomped_line = line.chomp
+    $log_file.puts("      #{chomped_line}")
+  }
+  raise("Error occurred in command:#{command}\nCommand log:#{command_res}") if command_res.match(/fatal|fail|error|aborted/i) if raise_error
```
Line is too long, hard to read code.

bbfsdev commented on bin/daily_tests.rb:
```diff
@@ -0,0 +1,221 @@
+    time_now = Time.now
+    puts("\nWake at #{time_now} and Start Daily test execution")
+    unless Gem::win_platform?
+      basic_dir = File.join('/tmp','daily_tests')
```
Consider taking base directory from parameters. I don't like /tmp/ is being used. I like being able to see the logs for last week or even month if needed.

bbfsdev commented on bin/daily_tests.rb:
```diff
@@ -0,0 +1,221 @@
+def unit_test
+
+  # init log
+  $log_file = File.open($DAILY_TEST_LOG_FILE, 'w')
```
Why not use the log gem?

bbfsdev commented on bin/daily_tests.rb:
```diff
@@ -0,0 +1,221 @@
+    while exec_index < 99999
+      dir_time_format = "#{time_now.year}_#{time_now.month}_#{time_now.day}.exec_#{exec_index}"
+      trial_file = File.join(basic_dir, dir_time_format)
+      break unless File.exist?(trial_file)
```
What is this loop? What is it purpose?

AlexeyNemytov committed:
1. ADD SERVER NAME TO LOG AND COMMAND LINE
2. ADD IIMIDIATE RUN COMMAND NOT TO WAIT FOR SLEEP
3. MKDIR FOR BBFS DIR IF NOT EXIST

bbfsdev commented:
What is the status of this?

bbfsdev commented:
Why is this not merged yet, what is the status of this?

bbfsdev commented:
Why is this not merged?
